painttiming.bs

```diff
@@ -186,21 +189,32 @@ Processing model {#sec-processing-model}
 Reporting paint timing {#sec-reporting-paint-timing}
 --------------------------------------------------------
 
+<h4 dfn>Previously reported paints</h4>
+Every [=Document=] has an associated [=set=] of previously reported paints, initiallized to an empty [=set=].
+
 <h4 dfn export>Mark paint timing</h4>
 
 <div algorithm="Mark paint timing">
     When asked to [=mark paint timing=] given a [=Document=] |document| as input, perform the following steps:
 
     1. Let |paintTimestamp| be the [=current high resolution time=].
-    1. If this instance of [=update the rendering=] is the [=first paint=] of |document|, then invoke the [[#report-paint-timing]] algorithm with |document|, <code>"first-paint"</code>, and |paintTimestamp|.
+    1. Let |reportedPaints| be the [=previously reported paints=] associated with |document|.
+    1. If this instance of [=update the rendering=] is the [=first paint=] of |document|, and |reportedPaints| does not contain <code>"first-paint"</code>, then:
+        1. invoke the [[#report-paint-timing]] algorithm with |document|, <code>"first-paint"</code>, and |paintTimestamp|.
+
+            NOTE: [=First paint=] excludes the default background paint, but includes non-default background paint.
 
-        NOTE: [=First paint=] excludes the default background paint, but includes non-default background paint.
+        1. [=Append=] <code>"first-paint"</code> to |reportedPaints|.
 
-    1. If this instance of [=update the rendering=] is the [=first contentful paint=] of |document|, then invoke the [[#report-paint-timing]] algorithm with |document|, <code>"first-contentful-paint"</code>, and |paintTimestamp| as arguments.
+    1. If this instance of [=update the rendering=] is the [=first contentful paint=] of |document|, and |reportedPaints| does not contain <code>"first-contentful-paint"</code>, then:
```

Comment:
I thought we wanted to fix this saying 'first contentful paint' and instead refer to contentful and paintable elements?